tests/test_list.py

```diff
@@ -129,24 +133,148 @@ class HasIntAList(properties.HasProperties):
         assert isinstance(deser_list[2], HasIntA) and deser_list[2].a == 100
 
         class HasOptionalList(properties.HasProperties):
-            mylist = properties.List('', properties.Bool(''), required=False)
+            mylist = list_class('', properties.Bool(''), required=False)
 
         hol = HasOptionalList()
         hol.validate()
 
         assert HasIntAList._props['mylist'].deserialize(None) is None
 
-        assert properties.List('', properties.Instance('', HasIntA)).equal(
+        assert list_class('', properties.Instance('', HasIntA)).equal(
             [HasIntA(a=1), HasIntA(a=2)], [HasIntA(a=1), HasIntA(a=2)]
         )
-        assert not properties.List('', properties.Instance('', HasIntA)).equal(
+        assert not list_class('', properties.Instance('', HasIntA)).equal(
             [HasIntA(a=1), HasIntA(a=2)],
             [HasIntA(a=1), HasIntA(a=2), HasIntA(a=3)]
         )
-        assert not properties.List('', properties.Instance('', HasIntA)).equal(
+        assert not list_class('', properties.Instance('', HasIntA)).equal(
             [HasIntA(a=1), HasIntA(a=2)], [HasIntA(a=1), HasIntA(a=3)]
         )
-        assert not properties.List('', properties.Integer('')).equal(5, 5)
+        assert not list_class('', properties.Integer('')).equal(5, 5)
+
+    def test_basic_vs_advanced(self):
```

Comment:
This test may be helpful in demonstrating how the new `List` fires change notifications.